src/main/java/de/tum/in/www1/artemis/service/metis/PostService.java

```diff
@@ -485,4 +558,150 @@ private void mayInteractWithPostElseThrow(Post post, User user, Course course) {
+            if (order != null) {
+                return order;
+            }
+        }
+
+        // sort by answer count
+        if (postSortCriterion == PostSortCriterion.ANSWER_COUNT) {
+            order = sortByAnswerCount(postA, postB, sortingOrder);
+            if (order != null) {
+                return order;
+            }
+        }
+        return 0;
+    }
+
+    @Nullable
+    private static Integer sortByPriority(Post postA, Post postB) {
+        if ((postA.getCourseWideContext() == CourseWideContext.ANNOUNCEMENT && postA.getDisplayPriority() == DisplayPriority.PINNED
+                && postB.getCourseWideContext() != CourseWideContext.ANNOUNCEMENT)
+                || ((postA.getDisplayPriority() == DisplayPriority.PINNED && postB.getDisplayPriority() != DisplayPriority.PINNED)
+                        || postA.getDisplayPriority() != DisplayPriority.ARCHIVED && postB.getDisplayPriority() == DisplayPriority.ARCHIVED)) {
+            return -1;
+        }
+        else if ((postA.getCourseWideContext() != CourseWideContext.ANNOUNCEMENT && postB.getCourseWideContext() == CourseWideContext.ANNOUNCEMENT
+                && postB.getDisplayPriority() == DisplayPriority.PINNED)
+                || (postA.getDisplayPriority() != DisplayPriority.PINNED && postB.getDisplayPriority() == DisplayPriority.PINNED)
+                || postA.getDisplayPriority() == DisplayPriority.ARCHIVED && postB.getDisplayPriority() != DisplayPriority.ARCHIVED) {
+            return 1;
+        }
+        else {
+            return null;
+        }
+    }
+
+    @Nullable
+    private static Integer sortByVotes(Post postA, Post postB, SortingOrder sortingOrder) {
+        int postAVoteEmojiCount = 0;
+        int postBVoteEmojiCount = 0;
+
+        if (postA.getReactions() != null) {
+            postAVoteEmojiCount = (int) postA.getReactions().stream().filter((Reaction reaction) -> reaction.getEmojiId().equals(VOTE_EMOJI_ID)).count();
+        }
+
+        if (postB.getReactions() != null) {
+            postBVoteEmojiCount = (int) postB.getReactions().stream().filter((Reaction reaction) -> reaction.getEmojiId().equals(VOTE_EMOJI_ID)).count();
+        }
+
+        if (postAVoteEmojiCount > postBVoteEmojiCount) {
+            return sortingOrder == SortingOrder.DESCENDING ? -1 : 1;
+        }
+        if (postAVoteEmojiCount < postBVoteEmojiCount) {
+            return sortingOrder == SortingOrder.DESCENDING ? 1 : -1;
+        }
+        return null;
+    }
+
+    @Nullable
+    private static Integer sortByCreationDate(Post postA, Post postB, SortingOrder sortingOrder) {
+        if (postA.getCreationDate().compareTo(postB.getCreationDate()) > 0) {
+            return sortingOrder == SortingOrder.DESCENDING ? -1 : 1;
+        }
+        if (postA.getCreationDate().compareTo(postB.getCreationDate()) < 0) {
+            return sortingOrder == SortingOrder.DESCENDING ? 1 : -1;
+        }
+        return null;
+    }
```

Comment:
This pattern is repeated a few times. I think something like
```suggestion
    private static int compareByCreationDate(Post postA, Post postB, SortingOrder sortingOrder) {
        int comparisonResult = 0;
        if (postA.getCreationDate().compareTo(postB.getCreationDate()) > 0) {
            comparisonResult = 1;
        }
        else if (postA.getCreationDate().compareTo(postB.getCreationDate()) < 0) {
            comparisonResult = -1;
        }
        return applySortingOrder(sortingOrder, comparisonResult);
    }

    private static int applySortingOrder(SortingOrder sortingOrder, int comparisonResult) {
        if (SortingOrder.ASCENDING == sortingOrder) {
            return comparisonResult;
        }
        else {
            return -1 * comparisonResult;
        }
    }
```
could reduce the number of branches and the duplication here and in the other helper methods.

Alternatively, the helper methods do not get the the `sortingOrder` parameter at all and instead the `postComparator` saves the relevant `comparisonResult` in a variable and only applies the sort order at the very end.